SU2_CFD/src/solvers/CTurbSolver.cpp

```diff
@@ -37,6 +37,11 @@ CTurbSolver::CTurbSolver(CGeometry* geometry, CConfig *config, bool conservative
   : CScalarSolver<CTurbVariable>(geometry, config, conservative) {
   /*--- Store if an implicit scheme is used, for use during periodic boundary conditions. ---*/
   SetImplicitPeriodic(config->GetKind_TimeIntScheme_Turb() == EULER_IMPLICIT);
+
+  implicit = (config->GetKind_TimeIntScheme_Turb() == EULER_IMPLICIT);
```

Comment:
having "implicit" has member is starting to sound error prone.
Keep muscl and limiter but get implicit where it is needed via config->GetKind_TimeIntScheme() as before

Reply:
Alright. But a differentiation between `GetKind_TimeIntScheme_Turb/Species()` has to be made, right? I currently do not see a major reason why the Turb and Species system should not be solvable with a different time integration scheme. Any thoughts

Reply:
That is why we have things like:
```
config[val_iZone]->SetGlobalParam(config[val_iZone]->GetKind_Solver(), RUNTIME_SPECIES_SYS);
```
To then be able to use the generic `GetKind_` without Turb/Species in "runtime function", it is only during construction that the specifics are needed.
So in fact we should also be able to remove the muscl and limiter member vars.

Reply:
I apologize if I do not understand it, but I am not sure whether you suggestion solves the problem.

**The Problem**
We have functions for e.g. the time integration scheme:
```c++
GetKind_TimeIntScheme_Turb/Species
GetKind_TimeIntScheme_Turb/Species
GetKind_TimeIntScheme_Turb/Species
```

Reply:
5fc376e implements your suggestion, Now limiter and muscl are local const bools which use the generic function 👍

Reply:
the `Preprocessing` routines actually get a RUNTIME_SYS as input so there it is no prob because one could call the SetGolbalParams it in the Preprocessing routine itself ... but the Turbulence solvers compute muT in Postprocessing which do not get  a `RUNTIME_SYS ` as input... so I guess right now limiter, turb and implicit are not used in the Postprocess routines of the turb models but that might become a nasty bug at some point